### PR TITLE
New UI file resolution

### LINKED ISSUE #101: "File resolution" breaks aspect ration
The **File Resolution** in options is very nice, because it allows you to reduce video size and increase encoding speed without actually changing your display resolution. However, currently it seems it hardcodes a few specific resolutions and doesn't really take into consideration that different aspect ratios might exist (like 4:3, or 16:10, or 21:9, or 32:10 with dual monitors, etc).

## Current Behavior
I have two 1920x1080 monitors, giving me 3840x1080 resolution. I needed to record the full area, but I wanted to downscale the final video. I picked **HD (720p)**. The ouput video was heavily distorted, because original aspect ratio was not kept, which means everything in the video was very tall and very slim, like this:
vlcsnap-2016-09-08-11h31m13s571

## Expected Behavior
I expected the video height to be roughly 720p, but the video width to be reflecting the aspect ratio of the original video.

## Possible Solution
I see two possible solutions:
1. Let the user set desired height (480, 720, 1080, etc) and compute the correct width from the recorded area.
2. Instead of hardcoding several pre-defined resolution, change this to a slider that allows downscaling the video. So I could move the slider and choose that the final video should be e.g. 50%, 65%, 80% of the recorded area size. That would allow finer control for everybody, and could be even more useful.

## Steps to Reproduce (for bugs)
1. set up two monitors
2. set **File Resolution** to e.g. `HD`
3. record both screens and see video distorted
4. (or if you have just a single monitor, just record a selected area and pick an area which is tall and narrow, that will also distort the picture quite visibly)

## Logs
I can provide logs if needed, I don't think there's anything important.

## Your Environment
* where you downloaded the extension?
 [x] Gnome shell extension website
 [ ] Last release from git repo
 [ ] Last version from git on master branch repo
* gnome shell version:
gnome-shell-3.21.90.1-1.fc25.x86_64
* Operating System and version:
Fedora 25

### PR DESCRIPTION
**Pull request by iacopodeenosee**
[as required](https://github.com/EasyScreenCast/EasyScreenCast/issues/101), i reviewed the selection of the screencast file, now the user can set a **native resolution**, choose one from a **list of preset** or set a **custom one** with the ability to maintain the aspect ratio or not; the UHD+ resolution made errors with vp8/vp9 encoding so i deleted them.

close #101

### CODE REVIEW

File: prefs.js [hunk 1]
```diff
@@ -276,13 +276,150 @@ const EasyScreenCastSettingsWidget = new GObject.Class({
...
             Settings.settings.bind(
-                Settings.FILE_RESOLUTION_SETTING_KEY, this.Ref_combobox_Resolution,
+                Settings.FILE_RESOLUTION_TYPE_SETTING_KEY,
+                this.Ref_combobox_Resolution,
                 'active', Gio.SettingsBindFlags.DEFAULT);
 
+            //intercept combobox res changed and update width/height value
+            this.Ref_combobox_Resolution.connect('changed',
+                Lang.bind(this, function(self) {
+                    var activeRes = self.active;
```
Comment (tobias47n9e) [inline prefs.js]: I wonder if we should add new code with `let` variables because of the clearer scope.

File: prefs.js [hunk 2]
```diff
@@ -276,13 +276,150 @@ const EasyScreenCastSettingsWidget = new GObject.Class({
...
+                'scl_ResHeight');
+            this.Ref_scale_HeightRes.set_valign(Gtk.Align.START);
+            this.Ref_scale_HeightRes.set_adjustment(adjustmentResHeight);
+            this.Ref_scale_HeightRes.set_digits(0);
+            this.Ref_scale_HeightRes.set_value(Settings.getOption(
+                'i', Settings.FILE_RESOLUTION_HEIGHT_SETTING_KEY));
+
+
+            // add marks on width/height file resolution
+            //let ind = 0;
```
Comment (tobias47n9e) [inline prefs.js]: Is `//let ind = 0;` still needed?

**Comment by tobias47n9e** [comment]
Sorry. Didn't see the review request. Will look at it in more hopefully soon.

**Comment by iacopodeenosee** [comment]
discard this PR because i continued the work, new tab info and support in the options window and other bugfix, and the two problems that you have reported out above I'll fix them in the next PR.

**Pull request closed without merging**